### PR TITLE
fix: use new magic link format

### PR DESCRIPTION
**Pull request by JazzarKarim**
Issue #: /https://github.com/bcgov/entity/issues/24363

- Fixed a lot of bugs with the magic link that've been occurring in DEV
- Refactored the flow
- Fixed unit tests

**New magic link format was done in:** https://github.com/bcgov/sbc-auth/pull/3166

### CODE REVIEW

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://business-registry-dev--pr-34-xhi8yxca.web.app

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://business-registry-dev--pr-34-xhi8yxca.web.app

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://business-registry-dev--pr-34-xhi8yxca.web.app

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://business-registry-dev--pr-34-xhi8yxca.web.app

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://business-registry-dev--pr-34-xhi8yxca.web.app

File: business-registry-dashboard/app/middleware/00.magic-link.global.ts [hunk 1]
```diff
@@ -0,0 +1,16 @@
+export default defineNuxtRouteMiddleware((to) => {
```
[the remark this answers is not in the document]
Comment (seeker25) [reply business-registry-dashboard/app/middleware/00.magic-link.global.ts:1]: do we need the locale prefix?

[the remark this answers is not in the document]
Comment (JazzarKarim) [reply business-registry-dashboard/app/middleware/00.magic-link.global.ts:1]: yep, we do

[the remark this answers is not in the document]
Comment (severinbeauvais) [reply business-registry-dashboard/app/middleware/00.magic-link.global.ts:1]: This looks concise and clear to me. (Love the comments.)

**Review by severinbeauvais — APPROVED** [review]
LGTM

**Review by deetz99 — APPROVED** [review]
👍

**Pull request merged**